File: octopoints_flutter/lib/db/dao/RuleDao.dart
import 'package:floor/floor.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/RuleEntity.dart';

@dao
abstract class RuleDao extends IDao<RuleEntity> {
  @Query('SELECT * FROM rules WHERE matchId=:id')
  Future<List<RuleEntity>> getRules(int id);
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/RuleDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.RuleEntity

@Dao
interface RuleDao : IDao<RuleEntity>{
    @Query("SELECT * FROM rules WHERE matchId=:id")
    suspend fun getRule(id: Int): RuleEntity?
}